pilot/pkg/security/authz/model/model.go

```diff
@@ -308,7 +315,15 @@ func replaceTrustDomainAliases(trustDomain string, trustDomainAliases []string,
 func getPrincipalsForAliases(trustDomainAliases []string, principal string) []string {
 	principalsForAliases := []string{}
 	for _, tdAlias := range trustDomainAliases {
-		principalsForAliases = append(principalsForAliases, replaceTrustDomainInPrincipal(tdAlias, principal))
+		tdWithAsteriskPrefix := getTrustDomain(principal)
+		// If the trust domain has a prefix * (e.g. *local from *local/ns/foo/ns/bar), generate principal
+		// with the prefix * trust domain, rather than the trust domain from the aliases list.
+		// The aliases list should contain fully qualified trust domain names (no prefix or suffix).
+		if suffixMatch(tdAlias, tdWithAsteriskPrefix) {
+			principalsForAliases = append(principalsForAliases, replaceTrustDomainInPrincipal(tdWithAsteriskPrefix, principal))
```

Comment:
This basically says, if tdWithAsteriskPrefix matches any of the tdAlias (including trustDomain), add the original principal (including the *), in addition, add modified principal with tdAliases. 

replaceTrustDomainInPrincipal(tdWithAsteriskPrefix, principal) is a no-op.